process/smartContract/process.go

```diff
@@ -480,13 +493,15 @@ func (sc *scProcessor) computeTotalConsumedFeeAndDevRwd(
 			for _, outTransfer := range outAcc.OutputTransfers {
 				sentGas += outTransfer.GasLimit
 			}
-			consumedGas -= sentGas
+			consumedGas = safeSubUint64(consumedGas, sentGas)
 		}
 	}
 
 	totalFee := core.SafeMul(tx.GetGasPrice(), consumedGas)
 
-	consumedGas -= builtInFuncGasUsed
+	if !senderInAnotherShard {
+		consumedGas = safeSubUint64(consumedGas, builtInFuncGasUsed)
```

Comment:
1. If I understand correctly, here you remove the builtin function gas cost, to compute the devFees since the devFees apply only to gas consumed by the SC call not also the builtin functions.
Can you use a different variable name here e.g consumedGasWithoutBuiltin otherwise it is confusing.

2. Why is the builtin function gas cost removed for senderInShard only? 
I assume if builtInFuncGasUsed > 0 then it means something has been spent here in this shard, so why don't we subtract it if sender is in different shard?
Or if we have a cross shard SC TX, is it never possible to spend gas on builtin functions on the destination shard?

Reply:
Yes for 1.

Reply:
It is never possible to use gas on destination SC. Only in sender sc for built in functions